web/src/main/java/org/cbioportal/web/PatientController.java

```diff
@@ -14,45 +21,86 @@
+        }
     }
 
-    @RequestMapping(value = "/studies/{studyId}/patients/{patientId}", method = RequestMethod.GET)
+    @RequestMapping(value = "/studies/{studyId}/patients/{patientId}", method = RequestMethod.GET,
+        produces = MediaType.APPLICATION_JSON_VALUE)
     @ApiOperation("Get a patient in a study")
-    public ResponseEntity<Patient> getPatientInStudy(@PathVariable String studyId,
-                                                            @PathVariable String patientId) {
+    public ResponseEntity<Patient> getPatientInStudy(
+        @ApiParam(required = true, value = "Study ID e.g. acc_tcga")
+        @PathVariable String studyId,
+        @ApiParam(required = true, value = "Patient ID e.g. TCGA-OR-A5J2")
+        @PathVariable String patientId) throws PatientNotFoundException {
 
-        throw new UnsupportedOperationException();
+        return new ResponseEntity<>(patientService.getPatientInStudy(studyId, patientId), HttpStatus.OK);
     }
 
-    @RequestMapping(value = "/patients/fetch", method = RequestMethod.POST)
+    @RequestMapping(value = "/patients/fetch", method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE,
+        produces = MediaType.APPLICATION_JSON_VALUE)
     @ApiOperation("Fetch patients by ID")
-    public ResponseEntity<List<Patient>> fetchPatients(@RequestParam(defaultValue = "SUMMARY") Projection projection,
-                                                                        @RequestBody List<PatientIdentifier> patientIdentifiers) {
+    public ResponseEntity<List<Patient>> fetchPatients(
+        @ApiParam("Level of detail of the response")
+        @RequestParam(defaultValue = "SUMMARY") Projection projection,
+        @ApiParam(required = true, value = "List of patient identifiers")
+        @RequestBody List<PatientIdentifier> patientIdentifiers) throws PageSizeTooBigException {
 
-        throw new UnsupportedOperationException();
-    }
+        if (patientIdentifiers.size() > PagingConstants.MAX_PAGE_SIZE) {
+            throw new PageSizeTooBigException(patientIdentifiers.size());
+        }
+
+        List<String> studyIds = new ArrayList<>();
+        List<String> patientIds = new ArrayList<>();
 
-    @RequestMapping(value = "/patients/query", method = RequestMethod.POST)
-    @ApiOperation("Query patients by example")
-    public ResponseEntity<List<Patient>> queryPatientsByExample(@RequestParam(defaultValue = "SUMMARY") Projection projection,
-                                                                                 @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_SIZE) Integer pageSize,
-                                                                                 @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_NUMBER) Integer pageNumber,
-                                                                                 @RequestBody Patient examplePatient) {
+        for (PatientIdentifier patientIdentifier : patientIdentifiers) {
```

Comment:
I see .. this is where the "parallel array" comes from .. the actual argument through the api *is* a pairing of (study,patient)